crates/typed-store/src/metrics.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use std::sync::Arc;
+
+use once_cell::sync::OnceCell;
+use prometheus::{
+    exponential_buckets, register_histogram_vec_with_registry,
+    register_int_counter_vec_with_registry, HistogramVec, IntCounterVec, Registry,
+};
+
+#[derive(Debug)]
+pub struct DBMetrics {
+    pub rocksdb_iter_latency_seconds: HistogramVec,
+    pub rocksdb_iter_bytes: HistogramVec,
+    pub rocksdb_get_latency_seconds: HistogramVec,
+    pub rocksdb_get_bytes: HistogramVec,
+    pub rocksdb_multiget_latency_seconds: HistogramVec,
+    pub rocksdb_multiget_bytes: HistogramVec,
+    pub rocksdb_put_latency_seconds: HistogramVec,
+    pub rocksdb_put_bytes: HistogramVec,
+    pub rocksdb_delete_latency_seconds: HistogramVec,
+    pub rocksdb_deletes: IntCounterVec,
+    pub rocksdb_batch_commit_latency_seconds: HistogramVec,
+    pub rocksdb_batch_commit_bytes: HistogramVec,
+}
+
+impl DBMetrics {
+    pub fn make_db_metrics(registry: &Registry) -> &'static Arc<DBMetrics> {
+        static ONCE: OnceCell<Arc<DBMetrics>> = OnceCell::new();
+        ONCE.get_or_init(|| Arc::new(DBMetrics::new(registry)))
+    }
+    pub(crate) fn new(registry: &Registry) -> Self {
+        DBMetrics {
+            rocksdb_iter_latency_seconds: register_histogram_vec_with_registry!(
+                "rocksdb_iter_latency_seconds",
+                "Rocksdb iter latency in seconds",
+                &["cf_name"],
+                exponential_buckets(1e-6, 2.0, 24).unwrap(),
```

Comment:
I have some code in `span_latency_prom.rs` that calculates the factor given the top and bottom latencies, I think that's a bit clearer, feel free to steal that and refactor into a common function

Reply:
I'll definitely check it out, although this feels pretty straight forward to read